### PR TITLE
cometSupply end-to-end acceptance; cometRepay patched in QuarkBuilder

### PR DESCRIPTION
**Pull request by fluffywaffles**

### CODE REVIEW

**Review by hayesgm — APPROVED** [review]

**Comment by hayesgm** [comment]
looks good to me-- we can separately follow up on the CI choking on not _too_ complex code.

File: test/builder/QuarkBuilderCometRepay.t.sol [hunk 1]
```diff
@@ -115,7 +126,7 @@ contract QuarkBuilderCometRepayTest is Test, QuarkBuilderTest {
             account: address(0xa11ce),
             nonceSecret: bytes32(uint256(12)),
             assetSymbols: Arrays.stringArray("USDC", "USDT", "LINK", "WETH"),
-            assetBalances: Arrays.uintArray(1e6, 0, 0, 0), // has 1 USDC
+            assetBalances: Arrays.uintArray(2e6, 0, 0, 0), // has 1 USDC
```
Comment (scott-silver) [inline test/builder/QuarkBuilderCometRepay.t.sol:129]: update comment?

`// has 2 USDC`

File: test/builder/QuarkBuilderCometRepay.t.sol [hunk 2]
```diff
@@ -69,14 +79,15 @@ contract QuarkBuilderCometRepayTest is Test, QuarkBuilderTest {
...
+            quote_()
         );
     }
 
     function testCometRepayNotEnoughPaymentToken() public {
         QuarkBuilder builder = new QuarkBuilder();
-        PaymentInfo.PaymentMaxCost[] memory maxCosts = new PaymentInfo.PaymentMaxCost[](1);
-        maxCosts[0] = PaymentInfo.PaymentMaxCost({chainId: 1, amount: 0.5e6}); // action costs 0.5 USDC
+        Quotes.NetworkOperationFee[] memory networkOperationFees = new Quotes.NetworkOperationFee[](1);
+        networkOperationFees[0] =
```
Comment (scott-silver) [inline test/builder/QuarkBuilderCometRepay.t.sol:89]: should this be passed to `quote_` on line 112?

**Review by hayesgm — APPROVED** [review]

File: test/builder/QuarkBuilderCometRepay.t.sol (hunk 2 above)
Comment (fluffywaffles) [reply test/builder/QuarkBuilderCometRepay.t.sol]: actually it can be deleted

**Pull request merged**